src/main/java/org/snomed/snowstorm/rest/ConceptController.java

```diff
@@ -495,6 +500,25 @@ public AsyncConceptChangeBatch getConceptBulkChange(@PathVariable String branch,
 
 		timer.checkpoint("Find children");
 
+		// For each child, determine if any its descendants are members of the passed-in
+		// refset
+		if (!refsetId.equals("")) {
+			// Calculate refset membership once, and pass in as a filter for each
+			// child-descendents query
+			Collection<Long> refsetMemberIds = eclQueryService.selectConceptIds("^" + refsetId,
+					versionControlHelper.getBranchCriteria(branch), branch, true, null, null).getContent();
+
+			for (ConceptMini child : children) {
+
+				if (eclQueryService.hasAnyResults("<< " + child.getConceptId(), branch,
```

Comment:
This is probably the most efficient way. I expect most of the time goes into fetching the refsetMemberIds above.
You could reuse the `BranchCriteria` created above rather than recreating it for every child.

Reply:
Ah, good point on the branchCriteria.   Change made.